core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -62,18 +82,57 @@ angular.module('oppia').controller('RteHelperModalController', [
       $scope.$broadcast('externalSave');
 
       var customizationArgsDict = {};
-      for (var i = 0; i < $scope.tmpCustomizationArgs.length; i++) {
-        var caName = $scope.tmpCustomizationArgs[i].name;
-        if (caName === 'video_id') {
-          var temp = $scope.tmpCustomizationArgs[i].value;
-          customizationArgsDict[caName] = (
-            extractVideoIdFromVideoUrl(temp.toString()));
-        } else {
-          customizationArgsDict[caName] = (
-            $scope.tmpCustomizationArgs[i].value);
+      let caName : string;
+      if ($scope.currentRteIsMathExpressionEditor) {
+        var svgFile = $scope.tmpCustomizationArgs[0].value.svgFile;
+        var svgFileName = $scope.tmpCustomizationArgs[0].value.svg_filename;
```

Comment:
I've added docstrings to explain this.
Basically, for the case of the math rich text components, we need to handle the saving of the generated SVG file here because the process of saving the SVG is asynchronous and the saving of SVG to the backend is to be done only after the user clicks on the save button. The saving of SVGs to the backend cannot be done in the math RTE editor
because the control is passed to this function as soon as the user clicks on the save button.
As soon as we click save , `$scope.$broadcast('externalSave');` this event is propogated and the the code returns from the customization modal immediately and doesn't wait. Hence we save it in this controller

Reply:
Also for the images, save to the backend happens before the save button is clicked and the control is still in the file-path-editor while the image is being saved to backend. This is not the case for math-expression-content-editor